src/App.tsx

```diff
@@ -22,73 +22,150 @@ enum SortType {
 }
 
 type ReorderOptions = {
-  sortType: SortType,
-  isReversed: boolean,
+  sortType: SortType;
+  isReversed: boolean;
 };
 
-// Use this function in the render method to prepare goods
 export function getReorderedGoods(
   goods: string[],
   { sortType, isReversed }: ReorderOptions,
 ) {
-  // To avoid the original array mutation
   const visibleGoods = [...goods];
 
-  // Sort and reverse goods if needed
-  // eslint-disable-next-line no-console
-  console.log(sortType, isReversed);
+  visibleGoods.sort((good1, good2) => {
+    switch (sortType) {
+      case SortType.ALPHABET:
+        return good1.localeCompare(good2);
+
+      case SortType.LENGTH:
+        return good1.length - good2.length;
+
+      default:
+        return 0;
+    }
+  });
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
 
   return visibleGoods;
 }
 
-// DON'T save goods to the state
-// type State = {
-//   isReversed: boolean,
-//   sortType: SortType,
-// };
-
 export const App: React.FC = () => {
+  const [reorderOptions, setReorderOptions] = useState({
+    sortType: SortType.NONE,
+    isReversed: false,
+  });
+
+  const reoderedGoods = getReorderedGoods(goodsFromServer, reorderOptions);
+
+  const sortAlphabetically = (e: React.MouseEvent<HTMLButtonElement>) => {
+    e.currentTarget.classList.remove('is-light');
+    document.querySelector('.buttons')?.children[1].classList.add('is-light');
+
+    setReorderOptions((state) => ({
+      ...state,
+      sortType: SortType.ALPHABET,
+    }));
+  };
+
+  const sortByLength = (e: React.MouseEvent<HTMLButtonElement>) => {
+    e.currentTarget.classList.remove('is-light');
+    document.querySelector('.buttons')?.children[0].classList.add('is-light');
+
+    setReorderOptions((state) => ({
+      ...state,
+      sortType: SortType.LENGTH,
+    }));
+  };
+
+  const reverse = (e: React.MouseEvent<HTMLButtonElement>) => {
+    e.currentTarget.classList.toggle('is-light');
+
+    setReorderOptions((state) => ({
+      ...state,
+      isReversed: !state.isReversed,
+    }));
+  };
+
+  const reset = () => {
+    document.querySelector('.buttons')?.children[0].classList.add('is-light');
+    document.querySelector('.buttons')?.children[1].classList.add('is-light');
+    document.querySelector('.buttons')?.children[2].classList.add('is-light');
```

Comment:
please try to rewrite it without using `document ...` (you can also change className with `useRef` hook)

Reply:
Done